machine/tool/clang/src/main/java/org/qbicc/machine/tool/clang/AbstractClangInvoker.java

```diff
@@ -56,29 +55,19 @@ void collectError(final Reader reader) throws IOException {
             Matcher matcher;
             while ((line = br.readLine()) != null) {
                 matcher = DIAG_PATTERN.matcher(line.trim());
-                ToolMessageHandler.Level level = ToolMessageHandler.Level.ERROR;
-                String fileOrExecutable = "";
-                String msg = line;
-                int lnum = -1;
                 if (matcher.matches()) {
-                    fileOrExecutable = matcher.group(1);
-                    try {
-                        lnum = Integer.parseInt(matcher.group(2));
-                    } catch (NumberFormatException e) {
-                        // just keep lnum at -1
-                    }
-                    String levelStr = matcher.group("levelStr");
-                    msg = matcher.group("msg");
-                    if (levelStr != null) {
-                        switch (levelStr) {
-                            case "note": level = ToolMessageHandler.Level.INFO; break;
-                            case "warning": level = ToolMessageHandler.Level.WARNING; break;
-                            case "error": level = ToolMessageHandler.Level.ERROR; break;
-                        }
+                    String levelStr = matcher.group(4);
+                    ToolMessageHandler.Level level;
+                    String msg;
+                    switch (levelStr) {
+                        case "note": level = ToolMessageHandler.Level.INFO; msg = matcher.group(5); break;
+                        case "warning": level = ToolMessageHandler.Level.WARNING; msg = matcher.group(5); break;
+                        case "error": level = ToolMessageHandler.Level.ERROR; msg = matcher.group(5); break;
+                        default: level = ToolMessageHandler.Level.ERROR; msg = levelStr; break;
```

Comment:
Better to format this in a blocks rather than as single lines
```suggestion
                        case "note": 
                            level = ToolMessageHandler.Level.INFO; 
                            msg = matcher.group(5); 
                            break;
                        case "warning":
                            level = ToolMessageHandler.Level.WARNING;
                            msg = matcher.group(5);
                            break;
                        case "error":
                            level = ToolMessageHandler.Level.ERROR;
                            msg = matcher.group(5);
                            break;
                        default:
                            level = ToolMessageHandler.Level.ERROR;
                            msg = levelStr;
                            break;
```